presto-hive/src/main/java/com/facebook/presto/hive/GenericHiveRecordCursor.java

```diff
@@ -117,6 +118,25 @@
     private Object rowData;
     private boolean closed;
 
+    public GenericHiveRecordCursor(
+            Configuration configuration,
+            Path path,
+            RecordReader<K, V> recordReader,
+            long totalBytes,
+            Properties splitSchema,
+            List<HiveColumnHandle> columns,
+            ZoneId hiveStorageTimeZoneId,
+            TypeManager typeManager)
+    {
+        this(configuration, path, recordReader, totalBytes, splitSchema, columns, getDateTimeZone(hiveStorageTimeZoneId), typeManager);
+    }
+
+    private static DateTimeZone getDateTimeZone(ZoneId hiveStorageTimeZoneId)
+    {
+        requireNonNull(hiveStorageTimeZoneId, "hiveStorageTimeZoneId is null");
+        return DateTimeZone.forTimeZone(java.util.TimeZone.getTimeZone(hiveStorageTimeZoneId));
```

Comment:
wouldn't `DateTimeZone.forID(hiveStorageTimeZoneId.getId())`  suffice here?
Also TimeZone.getTimeZone returns GMT for a zone if the string id cannot be understood.
See [javadoc](https://docs.oracle.com/javase/7/docs/api/java/util/TimeZone.html#getTimeZone(java.lang.String))

Reply:
Done